poethepoet/plugin.py

```diff
@@ -0,0 +1,77 @@
+import shutil
+import subprocess
+import sys
+from pathlib import Path
+from typing import Iterator
+
+import tomlkit
+
+try:
+    # pylint: disable=import-error
+    from cleo.commands.command import Command
+    from poetry.console import application as app
+    from poetry.plugins.application_plugin import ApplicationPlugin
+except ModuleNotFoundError:
+    # Hacky defaults to
+    # tell whether poetry is installed or not,
+    # please mypy,
+    # and please Poetry's plugin system (simulate the ApplicationPlugin object)
+    app = type("application", (), {"Application": None})  # pylint: disable=C0103
+    ApplicationPlugin = type(
```

Comment:
It looks like this try/except is just about making this module importable other than by poetry. Do you have a use case in mind for this?

Reply:
Good point. This module shouldn't be imported unless as a poetry plugin. Unless there is some wanted flexibility in the future? (But I guess we can always add it back)